design/notifications.md

```diff
@@ -0,0 +1,360 @@
+
+# Notifications
+This document outlines the design for notifications in Nessie with an application for Webhooks. It outlines the different
+events that we support, the way the notifications are called, how they are stored and what is the API used for creating,
+listing, updating and deleting notifications for a nessie server.
+
+## Event types
+This section lists out all events we support. All events in Nessie contain two pieces of information: the event time and
+the event type. The event time is the time at which the event was fired, it corresponds closely to when the event
+happened, but it can lag behind a few nanoseconds. The event type is an `UPPER_CASE` string representing the type of the
+event, as an example, commit events have a `"COMMIT"` type.
+
+For the following section, we will use the same schema as Nessie's API for referring to references, branches and tags,
+as well, as other Nessie API model objects, such as CommitMeta and Operation. 
+
+We will refer to them as their type in the Nessie OpenAPI spec, using the `$ref` keyword. 
+
+### Commits
+Commit events are triggered whenever there's a commit to a branch, i.e. whenever the `/trees/branch/{branchName}/commit`
+API in Nessie is called. 
+
+#### Schema
+```yaml
+CommitEvent:
+  type: object
+  properties:
+    type:
+      type: string
+      description: Always COMMIT
+    eventTime:
+      type: string
+      format: date
```

Comment:
I think this should be `date-time` (the same is used for `commitTime` in `CommitMeta`). Same comment applies to the other `eventTime` fields

Reply:
Probably mention above that all timestamps are in ISO format using UTC.